### PR TITLE
This change introduces a flag to not deploy extension during build. Default behavior will be true.

### PR DESCRIPTION
**Pull request by basoundr**
With the new VSSDK bits, build fails when we build the solution with deploy extension set to true in Jenkins. This change will help us pass Jenkins. Once this VSSDK issue gets fixed, we can remove this flag.

@dotnet/project-system @jinujoseph @mavasani @srivatsn for review

### CODE REVIEW

**Comment by basoundr** [comment]
@dotnet-bot  test ci please

**Comment by basoundr** [comment]
@mmitche Jenkins run did not start for this PR. Is there a Jenkins update that's happening now?

**Comment by basoundr** [comment]
@dotnet-bot test ci please

**Comment by mavasani** [comment]
@basoundr Tests will use the old netci.groovy for this PR. Once you merge the PR, it will generate new job configurations from the updated netci.groovy.

File: src/ProjectSystemDogfoodSetup/ProjectSystemDogfoodSetup.csproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <IncludeDebugSymbolsInVSIXContainer>false</IncludeDebugSymbolsInVSIXContainer>
     <IncludeDebugSymbolsInLocalVSIXDeployment>false</IncludeDebugSymbolsInLocalVSIXDeployment>
     <ImportVSSDKTargets>true</ImportVSSDKTargets>
+    <DeployExtension Condition="'$(DeployVsixExtension)' == 'false'">false</DeployExtension>
     <RestorePackages>true</RestorePackages>
```
Comment (davkean) [inline src/ProjectSystemDogfoodSetup/ProjectSystemDogfoodSetup.csproj]: Why don't you just set DeployExtension when you call MSBuild so you don't have to do this and it just automatically works?

**Comment by davkean** [comment]
Can you file a bug to revert this?

File: src/ProjectSystemDogfoodSetup/ProjectSystemDogfoodSetup.csproj (hunk 1 above)
Comment (basoundr) [inline src/ProjectSystemDogfoodSetup/ProjectSystemDogfoodSetup.csproj]: Setting `DeployExtension` to true makes msbuild think the non-VS projects(MS.VS.PS.Managed in this case) also to be vsix projects. It looks for vsixmanifest and fails. Although we specifically mention not to import VSSDK targets, setting the DeplyExtension property imports the VS SDK targets and eventually results in failure

**Comment by basoundr** [comment]
Will create an issue when I check this in. 

For now, still battling the Jenkins error. Currently GitHub seems to be down which is not letting the tests run properly.

**Comment by basoundr** [comment]
@dotnet-bot retest this please

**Comment by mmitche** [comment]
@basoundr What else are you seeing?

**Comment by 333fred** [comment]
@mmitche https://ci.dot.net/job/dotnet_roslyn-project-system/job/master/job/windows_debug_prtest/91/console

**Comment by mmitche** [comment]
@333fred That's pretty wacky.  Really weird.  Unfortunately I don't have the super detailed logs that going back that far, but it does look like something went wrong, because the usual environment variables weren't supplied (or were empty)

**Comment by basoundr** [comment]
I was running into https://ci.dot.net/job/dotnet_roslyn-project-system/job/master/job/windows_debug_prtest/91/console for quite a while. That seems to have gone.

**Comment by jinujoseph** [comment]
@dotnet-bot retest Windows Release please

**Comment by basoundr** [comment]
Opened #888 to track the reverting of the this commit

**Pull request merged**